waspc/src/Wasp/ExternalConfig.hs

```diff
@@ -0,0 +1,31 @@
+module Wasp.ExternalConfig
+  ( ExternalConfigs (..),
+    analyzeExternalConfigs,
+  )
+where
+
+import Control.Monad.Except (ExceptT (ExceptT), runExceptT)
+import StrongPath (Abs, Dir, Path')
+import Wasp.ExternalConfig.PackageJson (PackageJson, analyzePackageJsonContent)
+import Wasp.ExternalConfig.TsConfig (TsConfig, analyzeTsConfigContent)
+import Wasp.Project.Common
+  ( CompileError,
+    WaspProjectDir,
+  )
+
+data ExternalConfigs = ExternalConfigs
+  { packageJson :: PackageJson,
+    tsConfig :: TsConfig
+  }
+  deriving (Show)
+
+analyzeExternalConfigs :: Path' Abs (Dir WaspProjectDir) -> IO (Either [CompileError] ExternalConfigs)
```

Comment:
I've introduced a concept of "external configs" which are userland files which Wasp depends on but doesn't control. I'm not sure if something else would go into this category?